pkg/volume/cinder/cinder_util.go

```diff
@@ -224,6 +224,18 @@ func probeAttachedVolume() error {
 	scsiHostRescan()
 
 	executor := exec.New()
+
+	// udevadm settle waits for udevd to process the device creation
+	// events for all hardware devices, thus ensuring that any device
+	// nodes have been created successfully before proceeding.
+	argsSettle := []string{"settle", "--timeout=1"}
+	cmdSettle := executor.Command("udevadm", argsSettle...)
+	_, errSettle := cmdSettle.CombinedOutput()
+	if errSettle != nil {
+		glog.Errorf("error running udevadm settle %v\n", errSettle)
+		return errSettle
```

Comment:
I don't see any reference about errors in udevadm man page, but I don't think it should exit here (devices could settle later)

Reply:
The default for settle is 120 seconds. 1 second is still an eternity at runtime, but would be good to increase that or have some retries on here before giving up.

Reply:
@rootfs @stmcginnis Thanks very much. 
The latest commit has removed the exit for udevadm settle error,
and used the default settle command without --timeout.

Reply:
@edisonxiang, running the `udevadm settle` without the `--timeout=1` option will result in too many running `udevadm settle` processes and higher RAM and CPU usage. As mentioned in the #55007 , the func probeAttachedVolume() is run from WaitForAttach() every second, no matter if it returns anything or continues running. Therefore, the default udevadm timeout of 120 seconds makes no sense, and actually nothing longer than 1 second makes no sense either.

Reply:
@rootfs , i quite liked the original solution proposed by @edisonxiang with `udevadm settle --timeout=1`. The settle seems to be designed specifically for our case, and it does return non-zero exit code if udevd is busy processing the kernel message queue. The last command in the example given below returns exit code = 1. @stmcginnis, retries and timeouts are implemented in the calling functions.

```
# time udevadm settle ; echo $?
real    0m0.002s
user    0m0.002s
sys     0m0.000s
0

# for i in `seq 100`; do udevadm trigger; done & sleep 1; time udevadm settle ; echo $?
[1] 13262
[1]+  Done                    for i in `seq 100`;
do
    udevadm trigger;
done
real    0m11.074s
user    0m0.174s
sys     0m0.699s
0

# for i in `seq 100`; do udevadm trigger; done & sleep 1; time udevadm settle --timeout=1 ; echo $?
[1] 14106

real    0m1.046s
user    0m0.000s
sys     0m0.001s
1
```

Reply:
@kabakaev, I am not able to see where this code runs too many `udevadm settle` processes. Looking at `WaitForAttach` and `probeAttachedVolume`, they will run sequentially, I can't see any parallelism there.

Every 1, 2, 4, 8, ... seconds SCSI rescan happens, we wait until it's processed by udev and then check if the device is present. With `--timeout=1`, we don't give udev enough time to process old events and start a new SCSI rescan, which will load udev even more.

Reply:
@jsafrane , i stand corrected, the `WaitForAttach` indeed calls `probeAttachedVolume()` synchronously and waits for it to finish. So yes, the timeout of 120 seconds is reasonable. lgtm! 
Sorry for unnecessary notifications to edisonxiang, stmcginnis, and rootfs.

Reply:
Thanks @jsafrane.
Hi @kabakaev, hope this patch will resolve your issue.